File: Backend/Infrastructure.MyShop/Models/DTO/CategoryDTO/CategoryCreateDTO.cs
using Microsoft.AspNetCore.Http;

namespace Infrastructure.MyShop.Models.DTO.CategoryDTO
{
    public class CategoryCreateDTO
    {
        public string Name { get; set; }
        public IFormFile Image { get; set; }
        public string Description { get; set; }
        public int? ParentId { get; set; }
    }
}

File: Backend-Java/src/main/java/org/example/Infrastructure/dto/categoryDTO/CategoryCreateDTO.java
package org.example.Infrastructure.dto.categoryDTO;

import jakarta.validation.constraints.Null;
import lombok.Data;
import org.springframework.lang.Nullable;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;
import java.util.OptionalInt;

@Data //lombok {get; set;}
public class CategoryCreateDTO {
    private String name;
    private MultipartFile image;
    private String description;
    private Integer parentId;
}
